compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.h

```diff
@@ -126,6 +126,16 @@ class VectorLayoutAnalysis {
     return cast<T>(layout);
   }
 
+  /// Return the operation this on which this analysis was rooted on.
+  Operation *getRootOperation() const;
+
+  /// Duplicate layout information from the old value to the new value. This
+  /// is only expected to be used when cloning values. It is the job of the user
+  /// to ensure that there are no conflicts when cloning.
+  /// TODO: We should not rely on the user for this. The analysis should be
+  /// able to update itself and insert conflicts if needed.
+  void cloneLayoutInformationToNewValue(Value oldValue, Value newValue);
```

Comment:
You can use something like this from upstream to just do this maybe https://github.com/llvm/llvm-project/blob/7cc9ae95512edd0b969823fdfa062b92cb3c4d4e/mlir/lib/Dialect/Tensor/IR/TensorOps.cpp#L2941

Reply:
Actually, I don't need that anymore, so removing it.